XMonad/Hooks/DynamicIcons.hs

```diff
@@ -95,29 +95,43 @@ dynamicIconsPP ic pp = getWorkspaceIcons ic <&> \ren -> pp{ ppRename = ppRename
 -- | Returns a function for 'ppRename' that augments workspaces with icons
 -- according to the provided 'IconConfig'.
 getWorkspaceIcons :: IconConfig -> X (String -> WindowSpace -> String)
-getWorkspaceIcons IconConfig{..} = fmt <$> getWorkspaceIcons' iconConfigIcons
+getWorkspaceIcons IconConfig{..} = fmt <$> getWorkspaceIcons' iconFilterFunction iconConfigIcons
```

Comment:
At this point we might just pass the whole IconConfig to `getWorkspaceIcons'` … :-)